modules/light-clients/08-wasm/types/update_test.go

```diff
@@ -652,115 +661,99 @@ func (suite *TypesTestSuite) TestUpdateStateGrandpa() {
 // 	}
 // }
 
-/* func (suite *TypesTestSuite) TestUpdateStateOnMisbehaviourGrandpa() {
-	var (
-		ok          bool
-		clientMsg   exported.ClientMessage
-		clientState exported.ClientState
-	)
+func (suite *TypesTestSuite) TestUpdateStateOnMisbehaviour() {
+	var clientMsg exported.ClientMessage
 
 	testCases := []struct {
-		name    string
-		setup   func()
-		expPass bool
+		name               string
+		malleate           func()
+		panicErr           error
+		updatedClientState []byte
 	}{
 		{
-			"successful update",
+			"success: no update",
 			func() {
-				data, err := base64.StdEncoding.DecodeString(suite.testData["misbehaviour"])
-				suite.Require().NoError(err)
-				clientMsg = &types.ClientMessage{
-					Data: data,
-				}
+				suite.mockVM.RegisterSudoCallback(types.UpdateStateOnMisbehaviourMsg{}, func(_ wasmvm.Checksum, _ wasmvmtypes.Env, _ []byte, store wasmvm.KVStore, _ wasmvm.GoAPI, _ wasmvm.Querier, _ wasmvm.GasMeter, _ uint64, _ wasmvmtypes.UFraction) (*wasmvmtypes.Response, uint64, error) {
+					resp, err := json.Marshal(types.EmptyResult{})
+					if err != nil {
+						return nil, 0, err
+					}
 
-				clientState, ok = suite.chainA.App.GetIBCKeeper().ClientKeeper.GetClientState(suite.ctx, defaultWasmClientID)
-				suite.Require().True(ok)
+					return &wasmvmtypes.Response{Data: resp}, types.DefaultGasUsed, nil
+				})
 			},
-			true,
+			nil,
+			nil,
+		},
+		{
+			"success: client state updated on valid misbehaviour",
+			func() {
+				suite.mockVM.RegisterSudoCallback(types.UpdateStateOnMisbehaviourMsg{}, func(_ wasmvm.Checksum, _ wasmvmtypes.Env, _ []byte, store wasmvm.KVStore, _ wasmvm.GoAPI, _ wasmvm.Querier, _ wasmvm.GasMeter, _ uint64, _ wasmvmtypes.UFraction) (*wasmvmtypes.Response, uint64, error) {
+					// set new client state in store
+					store.Set(host.ClientStateKey(), types.MockClientStateBz)
```

Comment:
I mentioned also [here](https://github.com/cosmos/ibc-go/pull/4914/files#r1367964498), but maybe it's worth trying to unmarshal the payload from the `[]byte` argument instead of using directly `mockClientStateBz`, so that we can also check that the payload is marshalled correctly.